src/libstd/thread/hardware_threads.rs

```diff
@@ -0,0 +1,122 @@
+use crate::num::NonZeroUsize;
+
+/// Returns the number of hardware threads available to the process.
```

Comment:
Logically correct but in practice this should be smth like:
```
Returns the number of hardware threads available on CPU to OS.
```

Reply:
@vertexclique Can you elaborate? Are you trying to say this is the amount of threads available to the _operating system_ rather than the process?

This would've be accurate for non-OS VMs such as WASI runtimes, and process-based resource restriction for example through `cgroups`.

Reply:
Even in that scenario, existence of fork call makes more processes. So it might be a single process or not. That is basically ignoring the existence of fork and CommandExt.

Moreover, cgroups are not for single process. You can restrict a resource allocation to a single process or multiple processes. But if we are talking about cgroups then this will not show all exposed hardware threads to process/processes when cpuset is used.

+ One more thing, when thread affinity mask applied you won't get the all cores exposed to the process/processes. Same applies to programs using Numa and deployed with affinity mask applied.

Yet another thing is, after mask applied you will get vcores assigned to you. Not what OS gives then.

In the light of these, writing correct comment for this is important. It is not hardware threads, it is not os exposing and there are some other runtimes which is used as target arch and don't have OS.

Oh btw, check out sched_setaffinity for solving the last two problems.

Reply:
In addition to that, is this returning nvptx threads offloaded on gpu? Or any thread count(aka gpu lanes) per process/processes on GPUs like openmp does?

Reply:
@vertexclique Thanks for the explainer! What I've taken away is that there are many moving parts involved in determining how many threads are available to a program. Also it's hard to find a nomer that's both accurate and to the point.

I'm rewording the comment to use the word "program" rather than the more specific "process", and filing a note in the tracking issue that we need to do a pass on terminology before stabilization.

Reply:
This isn't just about cpusets/affinity. cgroups can also apply timeslice quotas. https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt
You probably don't want to run more threads than the quota since it would otherwise freeze your process every time the quota is exhausted early. I.e. background work could impact the interactive parts.